src/query/cypher_query_interpreter.cpp

```diff
@@ -146,8 +147,8 @@ std::shared_ptr<PlanWrapper> CypherQueryToPlan(uint64_t hash, AstStorage ast_sto
     }
   }
 
-  auto plan = std::make_shared<PlanWrapper>(
-      MakeLogicalPlan(std::move(ast_storage), query, parameters, db_accessor, predefined_identifiers));
+  auto plan = std::make_shared<PlanWrapper>(MakeLogicalPlan(std::move(ast_storage), query, parameters, db_accessor,
+                                                            std::move(db_acc), predefined_identifiers));
```

Comment:
Any plan which holds onto DatabaseAccessProtector should not be cached. Need to do analysis to say this query is not cachable.

Reply:
This likely isn't even needed if we just use ExecutionContext instead.